packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1230,6 +1230,47 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
     });
   }
 
+  /// Whether this has been disposed.
+  ///
+  /// If asserts are disabled, this property is always null.
+  bool? get debugDisposed {
+    bool? disposed;
+    assert(() {
+      disposed = _debugDisposed;
+      return true;
+    }());
+    return disposed;
+  }
+
+  bool _debugDisposed = false;
+
+  /// Release any resources held by this render object.
+  ///
+  /// If this render object has created any children directly, it should dispose
```

Comment:
should -> must dispose them before calling super.dispose() ? (since the method asserts that children are disposed?)

Reply:
Done